lib/config/index.js

```diff
@@ -1,37 +1,38 @@
 var nconf = require('nconf');
 var fs = require('fs');
 var path = require('path');
-var findRoot = require('find-root');
-var defaults = {};
-var getParentPath = function (parent) {
-        if (!parent  || !parent.filename) {
-            return null;
-        }
-
-        if (parent.filename.match(/ghost-ignition/gi)) {
-            return getParentPath(parent.parent);
-        }
-
-        // finds the root with package.json based on a path
-        return findRoot(parent.filename);
-    };
-var parentPath = getParentPath(module.parent);
-var env = require('./env');
-
-if (parentPath && fs.existsSync(path.join(parentPath, 'config.example.json'))) {
-    defaults = require(path.join(parentPath, 'config.example.json'));
-}
-
-nconf.set('NODE_ENV', env);
-
-nconf.argv()
-    .env()
-    .file({
-        file: path.join(parentPath, 'config.' + env + '.json')
-    });
-
-nconf.defaults(defaults);
-
-// @TODO: this file gets cached after the first require, we need to redesign how we load the config
-// @TODO: for example: require('ghost-ignition').config()
-module.exports = nconf;
+var debug = require('debug')('ignition:config');
+var config;
+
+var setupConfig = function setupConfig() {
+    var env = require('./env');
+    var defaults = {};
+    var parentPath = process.cwd();
+
+    config = new nconf.Provider();
+
+    if (parentPath && fs.existsSync(path.join(parentPath, 'config.example.json'))) {
+        defaults = require(path.join(parentPath, 'config.example.json'));
+    }
+
+    config.set('NODE_ENV', env);
```

Comment:
This won't work, because `config.env()` is called **after** this line.
So if `NODE_ENV` is undefined, `nconf` will always override your value with undefined.

Suggestion: move this line beneath `config.env()`:

```
// CASE: env is not set via env variable
if (config.get('NODE_ENV`))
  config.set('NODE_ENV', 'development')
```

Note: Using an env variable name can be confusing. That's why we use a custom variable in Ghost, see https://github.com/TryGhost/Ghost/blob/master/core/server/config/index.js#L41
Imagine you start Ghost without a NODE_ENV parameter, but the config always contains a value for this parameter. But that is not so important, just a tiny detail.